src/components/ReportActionItem/ReportActionItemImage.js

```diff
@@ -38,19 +40,35 @@ function ReportActionItemImage({thumbnail, image, enablePreviewModal}) {
     const imageSource = tryResolveUrlFromApiRoot(image || '');
     const thumbnailSource = tryResolveUrlFromApiRoot(thumbnail || '');
 
-    const receiptImageComponent = thumbnail ? (
-        <ThumbnailImage
-            previewSourceURL={thumbnailSource}
-            style={[styles.w100, styles.h100]}
-            isAuthTokenRequired
-            shouldDynamicallyResize={false}
-        />
-    ) : (
-        <Image
-            source={{uri: image}}
-            style={[styles.w100, styles.h100]}
-        />
-    );
+    // The source that we set for an Expensify Card eReceipt is `eReceipt/{transactionID}`
+    const isEReceipt = imageSource.startsWith(CONST.ERECEIPT_PATH);
```

Comment:
Alternative solution passing the transactionID in the image object for transactions.
I can't decide which method is better
cc @luacmartins 
```
diff --git a/src/components/ReportActionItem/MoneyRequestView.js b/src/components/ReportActionItem/MoneyRequestView.js
index ca1d99a174..35b1082232 100644
--- a/src/components/ReportActionItem/MoneyRequestView.js
+++ b/src/components/ReportActionItem/MoneyRequestView.js
@@ -153,6 +153,7 @@ function MoneyRequestView({report, betas, parentReport, policyCategories, should
                         <ReportActionItemImage
                             thumbnail={receiptURIs.thumbnail}
                             image={receiptURIs.image}
+                            transactionID={transaction.transactionID}
                             enablePreviewModal
                         />
                     </View>
diff --git a/src/components/ReportActionItem/ReportActionItemImage.js b/src/components/ReportActionItem/ReportActionItemImage.js
index e21ac53068..ae215cff5e 100644
--- a/src/components/ReportActionItem/ReportActionItemImage.js
+++ b/src/components/ReportActionItem/ReportActionItemImage.js
@@ -1,6 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import ONYXKEYS from '../../ONYXKEYS';
 import styles from '../../styles/styles';
 import Image from '../Image';
 import ThumbnailImage from '../ThumbnailImage';
@@ -12,6 +15,8 @@ import Navigation from '../../libs/Navigation/Navigation';
 import PressableWithoutFocus from '../Pressable/PressableWithoutFocus';
 import useLocalize from '../../hooks/useLocalize';
 import EReceiptThumbnail from '../EReceiptThumbnail';
+import transactionPropTypes from '../transactionPropTypes';
+import * as TransactionUtils from '../../libs/TransactionUtils';
 
 const propTypes = {
     /** thumbnail URI for the image */
@@ -22,10 +27,18 @@ const propTypes = {
 
     /** whether or not to enable the image preview modal */
     enablePreviewModal: PropTypes.bool,
+
+    /** The transactionID associated with this image, if any */
+    transactionID: PropTypes.string,
+
+    /* Onyx Props */
+    transaction: transactionPropTypes,
 };
 
 const defaultProps = {
     thumbnail: null,
+    transaction: {},
+    transactionID: '',
     enablePreviewModal: false,
 };
 
@@ -35,21 +48,18 @@ const defaultProps = {
  * and optional preview modal as well.
  */
 
-function ReportActionItemImage({thumbnail, image, enablePreviewModal}) {
+function ReportActionItemImage({thumbnail, image, enablePreviewModal, transaction, transactionID}) {
     const {translate} = useLocalize();
     const imageSource = tryResolveUrlFromApiRoot(image || '');
     const thumbnailSource = tryResolveUrlFromApiRoot(thumbnail || '');
-
-    // The source that we set for an Expensify Card eReceipt is `eReceipt/{transactionID}`
-    const isEReceipt = imageSource.startsWith(CONST.ERECEIPT_PATH);
+    const isEReceipt = !_.isEmpty(transaction) && TransactionUtils.isExpensifyCardTransaction(transaction) && TransactionUtils.hasEreceipt(transaction);
 
     let receiptImageComponent;
 
     if (isEReceipt) {
-        const transactionIDFromURL = imageSource.split(CONST.ERECEIPT_PATH)[1];
         receiptImageComponent = (
             <View style={[styles.w100, styles.h100]}>
-                <EReceiptThumbnail transactionID={transactionIDFromURL} />
+                <EReceiptThumbnail transactionID={transactionID} />
             </View>
         );
     } else if (thumbnail) {
@@ -97,4 +107,8 @@ ReportActionItemImage.propTypes = propTypes;
 ReportActionItemImage.defaultProps = defaultProps;
 ReportActionItemImage.displayName = 'ReportActionItemImage';
 
-export default ReportActionItemImage;
+export default withOnyx({
+    transaction: {
+        key: ({transactionID}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+    },
+})(ReportActionItemImage);
diff --git a/src/components/ReportActionItem/ReportActionItemImages.js b/src/components/ReportActionItem/ReportActionItemImages.js
index ce49f519df..c64e7a13b1 100644
--- a/src/components/ReportActionItem/ReportActionItemImages.js
+++ b/src/components/ReportActionItem/ReportActionItemImages.js
@@ -14,6 +14,7 @@ const propTypes = {
         PropTypes.shape({
             thumbnail: PropTypes.string,
             image: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+            transactionID: PropTypes.string,
         }),
     ).isRequired,
 
@@ -66,7 +67,7 @@ function ReportActionItemImages({images, size, total, isHovered}) {
 
     return (
         <View style={[styles.reportActionItemImages, hoverStyle, heightStyle]}>
-            {_.map(shownImages, ({thumbnail, image}, index) => {
+            {_.map(shownImages, ({thumbnail, image, transactionID}, index) => {
                 const isLastImage = index === numberOfShownImages - 1;
 
                 // Show a border to separate multiple images. Shown to the right for each except the last.
@@ -80,6 +81,7 @@ function ReportActionItemImages({images, size, total, isHovered}) {
                         <ReportActionItemImage
                             thumbnail={thumbnail}
                             image={image}
+                            transactionID={transactionID}
                         />
                         {isLastImage && remaining > 0 && (
                             <View style={[styles.reportActionItemImagesMoreContainer]}>
diff --git a/src/components/ReportActionItem/ReportPreview.js b/src/components/ReportActionItem/ReportPreview.js
index bfe5cc768c..b4cae18b3f 100644
--- a/src/components/ReportActionItem/ReportPreview.js
+++ b/src/components/ReportActionItem/ReportPreview.js
@@ -125,7 +125,6 @@ function ReportPreview(props) {
     const hasReceipts = transactionsWithReceipts.length > 0;
     const isScanning = hasReceipts && ReportUtils.areAllRequestsBeingSmartScanned(props.iouReportID, props.action);
     const hasErrors = hasReceipts && ReportUtils.hasMissingSmartscanFields(props.iouReportID);
-
     const lastThreeTransactionsWithReceipts = transactionsWithReceipts.slice(-3);
     const lastThreeReceipts = _.map(lastThreeTransactionsWithReceipts, (transaction) => ReceiptUtils.getThumbnailAndImageURIs(transaction));
 
diff --git a/src/libs/ReceiptUtils.ts b/src/libs/ReceiptUtils.ts
index 494b4b4d0a..90a90869cc 100644
--- a/src/libs/ReceiptUtils.ts
+++ b/src/libs/ReceiptUtils.ts
@@ -12,6 +12,7 @@ import ROUTES from '../ROUTES';
 type ThumbnailAndImageURI = {
     image: ImageSourcePropType | string;
     thumbnail: string | null;
+    transactionID?: string | null;
 };
 
 type FileNameAndExtension = {
@@ -34,7 +35,7 @@ function getThumbnailAndImageURIs(transaction: Transaction): ThumbnailAndImageUR
     const hasEReceipt = transaction?.hasEReceipt;
 
     if (hasEReceipt) {
-        return {thumbnail: null, image: ROUTES.ERECEIPT.getRoute(transaction.transactionID)};
+        return {thumbnail: null, image: ROUTES.ERECEIPT.getRoute(transaction.transactionID), transactionID: transaction.transactionID};
     }
 
     // For local files, we won't have a thumbnail yet


```

Reply:
I think using the transaction itself instead of the route is better. That being said, there's no need to pass both `transaction` and `transactionID`, I think all parents of `ReportActionItemImage(s)` already have access to the `transaction` object so we should just pass the `transaction` as a prop directly

Reply:
I tested that, but decided to go with `transactionID` instead due to the way the info is formatted/passed down from the helper function that gets the image source